test/unit/helpers/buyers/applications_helper_test.rb

```diff
@@ -1,66 +1,24 @@
+# frozen_string_literal: true
+
 require 'test_helper'
 
 class Buyers::ApplicationsHelperTest < ActionView::TestCase
 
-  test "services_contracted should return a json array of service ids" do
-    buyer = FactoryBot.create(:buyer_account)
-    service_ids = []
-    service_ids << FactoryBot.create(:service_contract, user_account: buyer).service.id
-    service_ids << FactoryBot.create(:service_contract, user_account: buyer).service.id
-    assert_same_elements service_ids, JSON.parse(services_contracted(buyer))
-  end
-
-  test "services_contracted should return a empty array" do
-    buyer = FactoryBot.create(:buyer_account)
-    assert_equal services_contracted(buyer), [].to_json
-  end
-
-  test "service_plan_contracted_for_service" do
-    buyer = FactoryBot.create(:buyer_account)
-    service_contract = FactoryBot.create(:service_contract, user_account: buyer)
-    buyer.reload
-
-    service_plan = buyer.bought_service_plans[0]
-
-    hash = {}
-    hash[service_plan.service.id] = {id: service_plan.id, name: service_plan.name}
-
-    assert_equal service_plan_contracted_for_service(buyer), hash.to_json
-  end
-
-  test "service_plan_contracted_for_service should return a empty hash" do
-    assert_equal service_plan_contracted_for_service(Account.providers.new), {}.to_json
+  test 'new_application_form_metadata requires a provider' do
+    assert_raise ArgumentError do
+      new_application_form_metadata
+    end
   end
 
-  test "relation_service_and_service_plans" do
+  test 'new_application_form_metadata' do
     provider = FactoryBot.create(:provider_account)
+    buyer = FactoryBot.create(:simple_buyer)
+    service = FactoryBot.create(:simple_service)
 
-    service = provider.services[0]
-    service_plan = service.service_plans[0]
-
-    hash = {}
-    hash[service.id] = [{id: service_plan.id, name: service_plan.name, default: false}]
-
-    assert_equal relation_service_and_service_plans(provider), hash.to_json
-  end
-
-  test "relation_service_and_service_plans should return a empty hash" do
-    assert_equal relation_service_and_service_plans(Account.providers.new), {}.to_json
-  end
-
-  test "relation_plans_services" do
-    application_plan = FactoryBot.create(:application_plan)
-    service = application_plan.service
-    provided_plan = service.provided_plans[0]
-    provider = provided_plan.provider_account
-
-    hash = {}
-    hash[application_plan.id] = service.id
-    assert_equal relation_plans_services(provider), hash.to_json
-  end
-
-  test "relation_plans_services should return a empty hash" do
-    assert_equal relation_plans_services(Account.providers.new), {}.to_json
+    assert new_application_form_metadata(provider: provider)
+    assert new_application_form_metadata(provider: provider, buyer: buyer)
+    assert new_application_form_metadata(provider: provider, service: service)
+    assert new_application_form_metadata(provider: provider, buyer: buyer, service: service)
```

Comment:
Could you compare returned values and make the test better understandable?
`assert` checks for truthy values not really useful here

Reply:
To be honest I don't even think these tests make any sense. It is impossible to compare with all possible return values.. IMO this is covered already with integration tests and cucumbers, since the React won't work unless the metadata is correct.

If you're OK with it I would remove it.. I just updated it because somebody added it long ago but like I said, I don't think it makes sense anymore.

Reply:
I did not see any integration test that makes use of this.
I saw:

- React Test but not calling the ruby method
- Cucumber test that tested one part of it not all possible values

If you think they do not make sense because there is too much data in them then  you could just stub `buyer_data`, `product_data` etc ... to make them smaller.

```ruby
buyer.stub(buyer_data: {foo: :bar})
product.stub(product_data: {greeting: :hello}
# etc...
```

Reply:
Cucumber test won't pass if the helper doesn't work correctly, because the React component won't render. I think it's enough with the cukes, it doesn't make sense to assert the return of a helper that's only supposed to arrange data in a certain way for the views.

Reply:
I am not a big fan of helpers arranging data ... Too much logic for a helper.
The goal of a helper is building something for the view without logic linked to the data.
Once you have to check a state of the data in the helper then it is not anymore a helper.

Thus putting it in a decorator or another pattern. And you can test this pattern object with stubs (to make your test easier)

Reply:
OK I moved it to the decorator and improved the tests a little bit